src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -1593,7 +1593,7 @@ internal Collection<PSObject> GetChildItems(string[] paths, bool recurse, uint d
                             }
 
                             int unUsedChildrenNotMatchingFilterCriteria = 0;
-                            ProcessPathItems(providerInstance, providerPath, recurse, context, out unUsedChildrenNotMatchingFilterCriteria, ProcessMode.Enumerate);
+                            ProcessPathItems(providerInstance, providerPath, recurse, context, out unUsedChildrenNotMatchingFilterCriteria, ProcessMode.Enumerate, depth: depth);
```

Comment:
I'd prefer to exclude parameters with defaults and use overloads.

Reply:
Just pushed new commit swapping optional parameter for overload. Is it along the right track?